src/contracts/SwapFactory.cdc

```diff
@@ -197,14 +193,14 @@ pub contract SwapFactory {
             )
 
             if self.lpTokenVaults.containsKey(pairAddr) {
-                let vaultRef = (&self.lpTokenVaults[pairAddr] as &FungibleToken.Vault?)!
+                let vaultRef = (&self.lpTokenVaults[pairAddr] as &{FungibleToken.Vault}?)!
                 vaultRef.deposit(from: <- lpTokenVault)
             } else {
                 self.lpTokenVaults[pairAddr] <-! lpTokenVault
             }
         }
 
-        pub fun withdraw(pairAddr: Address, amount: UFix64): @FungibleToken.Vault {
+        access(all) fun withdraw(pairAddr: Address, amount: UFix64): @{FungibleToken.Vault} {
```

Comment:
With the [addition of entitlements and safe downcasting,](https://github.com/onflow/flips/blob/main/cadence/20221214-auth-remodel.md) anyone with a public reference to a resource, such as a `&{SwapInterfaces.LpTokenCollectionPublic}` reference to this resource, can cast that reference to the full concrete interface type, which exposes all the `access(all)` functions and fields in the resource. This would mean that this `withdraw` function would be callable by anyone.

In order to restrict access to functions like this, you need to use [entitlements](https://cadence-lang.org/docs/1.0/language/access-control). 

I recommend that you read both of the documents that I linked so you can fully understand entitlements and where to use them. Not understanding and implementing these properly can result in critical bugs for your smart contract.

Reply:
@dryruner I would be careful not to mistake the difficulty and risk of learning the new Cadence features and upgrading the contracts for security issues inherent to Cadence 1.0. The language really hasn't changed much, so while some of the syntax may be a bit more verbose, the basic functionality is still all the same and the added restrictions that come with Capabilities actually make the language more powerful and more secure. There were so many very unintuitive things about learning and using access control and capabilities pre-1.0 that we take for granted now, but were a huge pain point for new learners that entitlements hopefully solve. It would have been great to have your feedback while we were designing these features to help us make it easier to use though! 

We probably should keep discussion here focused on your contracts though because the time for modifications to the Cadence 1.0 features has passed.

Reply:
The behavior of access modifiers and references is detailed in the FLIP I shared above, the access control docs page I shared above, and the [references](https://cadence-lang.org/docs/1.0/language/references) section of the docs site. Interfaces were never a good idea to use as access control, and allowing free upcasting and downcasting of references is a powerful feature. Instead of the opaque feature of access control to public functions being gated by which references the capability is created with, it is actually made explicit in the code how privileged functions can and should be accessed, which I would argue is more clear. Before, looking at the function declaration would not tell you which functions were meant to be public and private because they all were public.